File: day_6/palindrome.js
/**
 * palindrome - You are given a string. 
 * Write a function to check whether the string is a parlindrome or not.
 */

// Test cases

const a = 'kayak';
const b = 'custom';
const c = 'malayalam';
const d = 'a';
const e = 'aa';
const f = 'aA';

// brute method

function brutePalindrome (str) {
  const newStr = str.split('').reverse().join('');
  if (str === newStr) return true;
  return false;
}

// Second method

function palindromeB(str) {
  let newStr = [];
  for (let i = str.length - 1; i >= 0; i--) {
    newStr.push(str[i]);
  }
  newStr = newStr.join('');
  if (str === newStr) {
    return true;
  }
  return false;
}


// Using double pointer technique

function palindrome(str) {
  if (str.length === 0) return null;
  let flag;
  let i = 0;
  
  for (let j = str.length - 1; j >= 0; j--) {
    if (i < str.length && str[i] === str[j]) {
      flag = true;
      i++;
    } else {
      flag = false;
      return flag;
    }
  }
  return flag;
}



console.log(' ');
console.log('brute force');
console.log('');

console.log(brutePalindrome(a));
console.log(brutePalindrome(b));
console.log(brutePalindrome(c));
console.log(brutePalindrome(d));
console.log(brutePalindrome(e));
console.log(brutePalindrome(f));

console.log(' ');
console.log('Second method');
console.log('');

console.log(palindromeB(a));
console.log(palindromeB(b));
console.log(palindromeB(c));
console.log(palindromeB(d));
console.log(palindromeB(e));
console.log(palindromeB(f));


console.log(' ');
console.log('Two pointers method');
console.log('');

console.log(palindrome(a));
console.log(palindrome(b));
console.log(palindrome(c));
console.log(palindrome(d));
console.log(palindrome(e));
console.log(palindrome(f));

File: day_6/palindrome.py
"""
palindrome - You are given a string. 
Write a function to check whether the string is a parlindrome or not.
"""

# Test cases

a = 'kayak'
b = 'custom'
c = 'malayalam'
d = 'a'
e = 'aa'
f = 'aA'


# Brute fore method

def brute_palindrome(str):
  new_str = ''.join(reversed(str))
  if str == new_str:
    return True
  return False


# Second method

def palindrome_b(str):
  new_str = []
  for i in str:
    new_str.append(i)

  i = 0
  j = len(str) - 1

  while (i < j):
    new_str[i], new_str[j] = new_str[j], new_str[i]
    i+=1
    j-=1

  new_str = ''.join(new_str)
  if (str == new_str):
    return True
  else:
    return False


# Double poiner method

def palindrome(str):
  str_len = len(str) - 1

  i = 0
  j = str_len
  flag = True

  while (i <= j):
    if str[i] == str[j]:
      flag
    else:
      flag = False
      return flag
    i += 1
    j -= 1
  return flag




print(' ')
print('brute force')
print('')

print(brute_palindrome(a))
print(brute_palindrome(b))
print(brute_palindrome(c))
print(brute_palindrome(d))
print(brute_palindrome(e))
print(brute_palindrome(f))

print(' ')
print('Second method')
print('')

print(palindrome_b(a))
print(palindrome_b(b))
print(palindrome_b(c))
print(palindrome_b(d))
print(palindrome_b(e))
print(palindrome_b(f))


print(' ')
print('Two pointer method')
print('')

print(palindrome(a))
print(palindrome(b))
print(palindrome(c))
print(palindrome(d))
print(palindrome(e))
print(palindrome(f))